frontend/src/pages/Author.jsx

```diff
@@ -56,8 +68,169 @@ class Author extends Component {
+            requireAuth = true;
+        HTTPFetchUtil.getRequest(urlPath, requireAuth)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        for (var i = 0; i < results.authors.length; i++) {
+                            if (results.authors[i].id === authorID[0]){
+                                this.setState({
+                                    isFollowing: true
+                                })
+                            }
+                        }
+                    })
+                }
+            })
+            .catch((error) => {
+                console.error(error);
+        });    
+        } 
+
+    sendUnfollowRequest() {
+        var authorID;
+        authorID = this.getloggedinAuthorIDandHost()
+        let urlPath = "/api/unfollow/"
+		let body = {
+			query: "unfollow",
+			author: {
+				id: authorID[0],
+				host: authorID[1],
+			},
+			friend:{
+				id: this.props.location.state.fullAuthorId,
+				host: this.props.location.state.host,
+			}
+		}
+		HTTPFetchUtil.sendPostRequest(urlPath, true, body)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        this.setState({
+                            isFollowing: false
+                        })
+						toast(
+							{
+								type: 'success',
+								icon: 'user',
+								title: 'Unfollow successful!',
+								description: <p>You are now following {this.props.location.state.displayName}</p>
+							}
+						);
+
+                    })
+				}
+				else{
+					console.log(httpResponse)
+					toast(
+						{
+							type: 'error',
+							icon: 'warning',
+							title: 'Error: User could not be unfollowed!'
+						}
+					);
+				}
+            })
+            .catch((error) => {
+                console.error(error);
+        });
+    }
+
+    sendFollowRequest() {
+        var authorID;
+        authorID = this.getloggedinAuthorIDandHost()
+		let urlPath = "/api/friendrequest/"
+		let body = {
+			query: "friendrequest",
+			author: {
+				id: authorID[0],
+				host: authorID[1],
+			},
+			friend:{
+				id: this.props.location.state.fullAuthorId,
+				host: this.props.location.state.host,
+			}
+		}
+		HTTPFetchUtil.sendPostRequest(urlPath, true, body)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        this.setState({
+                            isFollowing: true
+                        })
+						toast(
+							{
+								type: 'success',
+								icon: 'user',
+								title: 'Follow successful!',
+								description: <p>You are now following {this.props.location.state.displayName}</p>
+							}
+						);
+
+                    })
+				}
+				else{
+					console.log(httpResponse)
```

Comment:
remove the console.log if you don't need this anymore

Reply:
ok